### DISCUSSION TITLE
Image component not showing in Google Cache System

### DISCUSSION THREAD

**Discussion by sefatunckanat** [vercel/next.js#39801] [Help]
Hello, i was deploy my first next project. We sent the site to google and it took it to cache. But we have few mistakes. 

### Image Component

Google could not cache any image. Images has a Image component and source is svg. I didn't use an extra config for images. 

*Google Cache*
<img width="1290" alt="image" src="https://user-images.githubusercontent.com/9070463/185797423-a33691fd-f121-4886-be6a-128c1814230e.png">

*Orginal*
<img width="1014" alt="image" src="https://user-images.githubusercontent.com/9070463/185797525-4ae2ed3f-7f61-4bc8-986e-447c76b8fdfd.png">

*Source*
<img width="203" alt="image" src="https://user-images.githubusercontent.com/9070463/185797786-670396e6-ee53-4c6e-a7c3-970066b2fcd3.png">

```
<div className={styles.bannerImage}>
<Image
  width={611}
  height={587}
  src="/images/BannerImage.svg"
  alt="Banner Image"
  objectPosition={"right"}
/>
</div>
```

### Global sass
I used module css. And Google has cached it. However, it did not cache the font styles in the sass file that I called globally. Can you help me where am I going wrong?

<img width="963" alt="image" src="https://user-images.githubusercontent.com/9070463/185798210-a702a818-71eb-40dd-8600-59b81bd9681a.png">

**Comment by freislot** [comment]
Same problem

**Comment by ezovsa** [comment]
I use next 12 and the same problem

**Comment by sefatunckanat** [comment]
Solved. But I think it's the wrong way. This affects the opening speed a lot.

`priority` property should be added to images. Next image shows a transparent image first. Then it downloads the source. Google also captures this transparent picture.

https://nextjs.org/docs/api-reference/next/image#priority

**Reply by freislot** [reply]
Thanks. We will try this prop

**Reply by ezovsa** [reply]
Thank you!

**Comment by ermakov95** [comment]
I have the same problem. It's strange that Next.js there is no built-in solution. 
I need all the images on the site to be displayed in webcache.googleusercontent. If I apply priority, it will kill lazy-loading.

Another discussion of this problem: https://github.com/vercel/next.js/discussions/37674

I figured out a little bit how it works. html comes like this:
![image](https://user-images.githubusercontent.com/89069249/226325217-c6a0687c-aee1-4e45-8937-526bb32823dd.png)
Initially there is an img tag, but with a strange src and there is a noscript tag with a normal img inside.

![image](https://user-images.githubusercontent.com/89069249/226322414-20129eaa-8a34-4fcb-97ed-c659e0b4ee03.png)
Then Next.js deletes the contents of the noscript tag and fills in the src in img and the picture is loaded and displayed.

However, in webcache.googleusercontent.com this does not happen and the html Image is output in its original form, so the image is not displayed.
![image](https://user-images.githubusercontent.com/89069249/226323177-6c194867-93b1-42bd-b0c0-54da6c8bd8a2.png)

If on webcache.googleusercontent.com turn off JS, then the images will be displayed from noscript.

"priority" solves this problem, but removes lazy-loading

**Reply by sefatunckanat** [reply]
You should try newest version of next.js. Many html props have been removed.

**Reply by shamoon1997** [reply]
I am facing the same issue. Nextjs image componet is not being cached. So, it always displays empty images. Anyone who has find a workaround for this?

**Comment by realamirhe** [comment]
It is not the Next (Image) component issue, Router issue prevent the full load for hydration process
```bash
DOMException: Failed to execute 'replaceState' on 'History': A history state object with URL 
```
